system/jlib/jsocket.cpp

```diff
@@ -394,10 +394,10 @@ static int keepAliveProbes = 9;
 
 static bool queryKeepAlive(int &time, int &intvl, int &probes)
 {
-    CriticalBlock block(queryKACS);
-
     if (doKeepAlive == UNINIT)
```

Comment:
this change can cause problems (races).
Another thread could call queryKeepAlive whilst a thread is in the CS block and has just set DISABLED for example, meaning the other thread will return false spuriously.
I think it would be ok, if you ensure that doKeepAlive set is at the end of the block, after any initialization is checked/done.

Reply:
Yes, thanks, coded it in too much of a hurry, added the extra if check.

Reply:
It's still an issue. e.g.:
Thread 1 (T1), gets to line 415.
Thread 2 (T2), hits line 397, and doKeepAlive is DISABLED , T2 incorrectly returns false.

>if you ensure that doKeepAlive set is at the end of the block, after any initialization is checked/done.

still think you need to do this, ensure it's not changed from UNINIT until after initialization is done (or fails), then other threads will enter the crit block.

Reply:
Agreed